src/utils/init/config-github.js

```diff
@@ -1,188 +1,215 @@
+    return githubToken.token
   }
-  const octokit = new Octokit({
-    auth: `token ${ghtoken.token}`,
-  })
-
-  const key = await api.createDeployKey()
-  const parsedURL = parseGitRemote(repo.repo_path)
 
-  await octokit.repos.addDeployKey({
-    title: 'Netlify Deploy Key',
-    key: key.public_key,
-    repo: parsedURL.name,
-    owner: parsedURL.owner,
-    read_only: true,
+  const newToken = await ghauth({
+    log,
   })
+  globalConfig.set(`users.${userId}.auth.github`, newToken)
+  return newToken.token
+}
 
-  repo.deploy_key_id = key.id
-
-  // TODO: Look these up and default to the lookup order
+const getGitHubClient = ({ token }) => {
+  const octokit = new Octokit({
+    auth: `token ${token}`,
+  })
+  return octokit
+}
 
-  let defaultBuildCmd
-  let defaultBuildDir = '.'
-  // read from netlify toml
-  const { build } = ctx.netlify.config
-  if (build && build.command) {
-    defaultBuildCmd = build.command
-  }
-  if (build && build.publish) {
-    defaultBuildDir = build.publish
+const addDeployKey = async ({ log, api, octokit, repoOwner, repoName, failAndExit }) => {
+  log('Adding deploy key to repository...')
+  const key = await createDeployKey({ api, failAndExit })
+  try {
+    await octokit.repos.addDeployKey({
+      title: 'Netlify Deploy Key',
+      key: key.public_key,
+      owner: repoOwner,
+      repo: repoName,
+      read_only: true,
+    })
+    log('Deploy key added!')
+    return key
+  } catch (error) {
+    let message = formatErrorMessage({ message: 'Failed adding GitHub deploy key', error })
+    if (error.status === 404) {
+      const { name, owner } = formatRepoAndOwner({ repoName, repoOwner })
+      message = `${message}. Does the repository ${name} exist and do ${owner} has the correct permissions to set up deploy keys?`
+    }
+    failAndExit(message)
   }
-  if (build && build.functions) console.log(`Netlify functions folder is ${chalk.yellow(build.functions)}`)
-  const { buildCmd, buildDir } = await inquirer.prompt([
-    {
-      type: 'input',
-      name: 'buildCmd',
-      message: 'Your build command (hugo build/yarn run build/etc):',
-      filter: (val) => (val === '' ? '# no build command' : val),
-      default: defaultBuildCmd,
-    },
-    {
-      type: 'input',
-      name: 'buildDir',
-      message: 'Directory to deploy (blank for current dir):',
-      default: defaultBuildDir,
-    },
-  ])
+}
 
-  const tomlpath = path.join(ctx.netlify.site.root, 'netlify.toml')
-  const tomlDoesNotExist = !fs.existsSync(tomlpath)
-  if (tomlDoesNotExist && (!ctx.netlify.config || Object.keys(ctx.netlify.config).length === 0)) {
-    const { makeNetlifyTOML } = await inquirer.prompt([
-      {
-        type: 'confirm',
-        name: 'makeNetlifyTOML',
-        message: 'No netlify.toml detected. Would you like to create one with these build settings?',
-        default: true,
-      },
-    ])
-    if (makeNetlifyTOML && ctx.netlify.site && ctx.netlify.site.root) {
```

Comment:
There are 3 issues here:
1. If the user declines to create `netlify.toml` they'll get an error.
2. The `ctx.netlify.site && ctx.netlify.site.root` is always true
3. The error message is not really helpful